src/metabase/driver/h2.clj

```diff
@@ -185,12 +185,33 @@
         (throw (ex-info "IllegalArgument: DDL commands are not allowed to be used with h2."
                         {:classification query-classification}))))))
 
+(defn- read-only-statements? [{:keys [command-types remaining-sql]}]
+  (let [cmd-type-nums command-types]
+    (boolean
+     (and (every? #{CommandInterface/SELECT ; includes SHOW, TABLE, VALUES
+                    CommandInterface/EXPLAIN
+                    CommandInterface/CALL} cmd-type-nums)
+          (nil? remaining-sql)))))
+
+(defn- check-read-only-statements [{:keys [database] {:keys [query]} :native}]
+  (when query
+    (let [query-classification (classify-query database query)]
+      (when-not (read-only-statements? query-classification)
+        (throw (ex-info "Only SELECT statements are allowed in a native query."
+                        {:classification query-classification}))))))
+
 (defmethod driver/execute-reducible-query :h2
   [driver query chans respond]
   (check-native-query-not-using-default-user query)
-  (check-disallow-ddl-commands query)
+  (check-read-only-statements query)
   ((get-method driver/execute-reducible-query :sql-jdbc) driver query chans respond))
 
+(defmethod driver/execute-write-query! :h2
```

Comment:
You mentioned implementing a read-only connection to address the `ACCESS_MODE_DATA` concerns. Is there anything that ensures that this uses a read-only connection? I looked at the `:sql-jdbc` implementation but didn't see anything.

Reply:
Obviously, not a read-only connection for the write query... excuse me 🤦 But wherever that would be implemented.

Reply:
The OP comment of the PR notes that dual connections is impossible, see https://github.com/metabase/metabase/pull/28377#issuecomment-1438610617 .

That's an unfortunate design from H2 but not something we can change in the near term.

Reply:
I see. Then, yeah, I'm out of my depth on the connection parameter question.